packages/jira-adapter/src/filters/assets/assets_object_type_order.ts

```diff
@@ -0,0 +1,196 @@
+const log = logger(module)
+
+const deployOrderChange = async ({
+  client, workspaceId, position, toObjectTypeId, assetsObjectType,
+}: {
+  client: JiraClient
+  workspaceId: string
+  position: number
+  toObjectTypeId?: string
+  assetsObjectType: ReferenceExpression
+}):
+Promise<void> => {
+  const { id } = assetsObjectType.value.value
+  const data = toObjectTypeId === undefined ? { position } : { position, toObjectTypeId }
+  const url = `/gateway/api/jsm/assets/workspace/${workspaceId}/v1/objecttype/${id}/position`
+  await client.post({ url, data })
+}
+
+const createOrderType = (): ObjectType => new ObjectType({
+  elemID: new ElemID(JIRA, OBJECT_TYPE_ORDER_TYPE),
+  fields: {
+    objectTypes: {
+      refType: new ListType(BuiltinTypes.NUMBER),
+      annotations: { [CORE_ANNOTATIONS.CREATABLE]: true, [CORE_ANNOTATIONS.UPDATABLE]: true },
+    },
+    assetsSchema: {
+      refType: BuiltinTypes.NUMBER,
+      annotations: { [CORE_ANNOTATIONS.CREATABLE]: true, [CORE_ANNOTATIONS.UPDATABLE]: true },
+    },
+  },
+  path: [JIRA, adapterElements.TYPES_PATH, OBJECT_TYPE_ORDER_TYPE],
+  annotations: {
+    [CORE_ANNOTATIONS.CREATABLE]: true,
+    [CORE_ANNOTATIONS.UPDATABLE]: true,
+    [CORE_ANNOTATIONS.DELETABLE]: true,
+  },
+})
+
+const createAssetsObjectTypeOrder = (
+  assetsObjectTypes: InstanceElement[],
+  orderType: ObjectType,
+  treeParent: InstanceElement
+): InstanceElement | undefined => {
+  const schema = treeParent.elemID?.typeName === OBJECT_TYPE_TYPE
+    ? treeParent.annotations[CORE_ANNOTATIONS.PARENT]?.[0]
+    : new ReferenceExpression(treeParent.elemID, treeParent)
+  if (schema.elemID?.typeName !== OBJECT_SCHEMA_TYPE) {
+    log.error(`Failed to create ${OBJECT_TYPE_ORDER_TYPE} for ${treeParent.elemID.getFullName()} because it's parent is not ${OBJECT_SCHEMA_TYPE}`)
+    return undefined
+  }
+  const name = naclCase(`${treeParent.value.name}_order`)
+  const subFolder = treeParent.elemID.typeName === OBJECT_TYPE_TYPE ? ['childOrder'] : ['assetsObjectTypes', 'childOrder']
+  return new InstanceElement(
+    name,
+    orderType,
+    {
+      objectTypes: assetsObjectTypes.sort((a, b) => a.value.position - b.value.position)
+        .map(inst => new ReferenceExpression(inst.elemID, inst)),
+      assetsSchema: schema,
+    },
+    [...(treeParent.path ?? []).slice(0, -1), ...subFolder, pathNaclCase(name)],
+    {
+      [CORE_ANNOTATIONS.PARENT]: [new ReferenceExpression(treeParent.elemID, treeParent)],
+    }
+  )
+}
+
+/* Handles the assetsObjectTypes order inside each assets objectType
+ * by creating an InstanceElement of the assetsObjectTypes order inside the assets objectType.
+ */
+const filterCreator: FilterCreator = ({ config, client, fetchQuery }) => ({
+  name: 'assetsObjectTypeOrderFilter',
+  onFetch: async (elements: Element[]) => {
+    if (!config.fetch.enableJSM
+    || !config.fetch.enableJsmExperimental
+    || !fetchQuery.isTypeMatch(OBJECT_TYPE_ORDER_TYPE)) {
```

Comment:
maybe - 
```suggestion
    || !fetchQuery.isTypeMatch(OBJECT_TYPE_TYPE)) {
```

see my comment above 🙏